### ISSUE TITLE
type.__subclasses__() incorrectly inferred with bound type variant

### ISSUE THREAD

**Issue by BlueGlassBlock** [microsoft/pyright#3897]
**Describe the bug**
When a type variant is bound to a certain type, its `__subclasses__()` will no longer include the "type variance" information, causing it unable to assign to original type variant.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import TypeVar


class MyCls:
    ...


T = TypeVar("T", bound=MyCls)


def gen_subclass_limited(cls: type[T]) -> list[type[T]]:
    result = [cls]
    for c in cls.__subclasses__():
        result.extend(gen_subclass_limited(c)) 
    # Argument of type "list[Type[MyCls]*]*" cannot be assigned to parameter "__iterable" of type "Iterable[Type[T@gen_subclass]]" in function "extend"
  
    # TypeVar "_T_co@Iterable" is covariant

    # Type "Type[MyCls]*" cannot be assigned to type "Type[T@gen_subclass]"
    return result

# However this passes checks...

M = TypeVar("M")


def gen_subclass(cls: type[M]) -> list[type[M]]:
    result = [cls]
    for c in cls.__subclasses__():
        result.extend(gen_subclass(c))
    return result

```

**Expected behavior**
The code passes check.

**VS Code extension or command-line**
Pylance v2022.8.51

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.270, which I just published. It will also be included in a future release of pylance.
